aws_lambda/aws_lambda.py

```diff
@@ -432,6 +453,35 @@ def update_function(cfg, path_to_zip_file):
 
     client.update_function_configuration(**kwargs)
 
+def upload_s3(cfg, path_to_zip_file):
+    """Upload a function to AWS S3."""
+
+    print('Uploading your new Lambda function')
+    byte_stream = read(path_to_zip_file, binary_file=True)
+    aws_access_key_id = cfg.get('aws_access_key_id')
+    aws_secret_access_key = cfg.get('aws_secret_access_key')
+
+    account_id = get_account_id(aws_access_key_id, aws_secret_access_key)
+    role = get_role_name(account_id, cfg.get('role', 'lambda_basic_execution'))
+
+    client = get_client('s3', aws_access_key_id, aws_secret_access_key,
+                        cfg.get('region'))
+
+    # Do we prefer development variable over config?
+    buck_name = (
+        os.environ.get('S3_BUCKET_NAME') or cfg.get('bucket_name')
+    )
+    func_name = (
+        os.environ.get('LAMBDA_FUNCTION_NAME') or cfg.get('function_name')
+    )
+    kwargs = {
+        'Bucket': '{}'.format(buck_name),
+        'Key': cfg.get('s3_key', '{}.zip'.format(func_name)),
```

Comment:
this should be generated programmatically, maybe like a unix timestamp and the md5 of the the zip file. The configuration should allow for the path to be specified.

e.g.:
```python

# this is untested
s3_key_prefix = cfg.get('s3_key_prefix', '/dist')
checksum = md5(btye_stream).digest()
timestamp = str(time.time())
filename = '{prefix}{checksum}-{ts}.zip'.format(prefix=s3_key_prefix, checksum=checksum, ts=timestamp)
```